### ISSUE TITLE
Support creating a model with paper programs

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/papyrus#21]
We started to look at how to create a model with paper programs. Our first pass encodes the model like this in a program:

```js
  await paper.set( 'data', {
    model: {
      flyingThingAltitudeRange: {
        type: 'dot.Range',
        args: [ 0, 1 ]
      },
      flyingThingAltitudeProperty: {
        type: 'axon.NumberProperty',
        args: [ 0, 'model.flyingThingAltitudeRange' ]
      },

      // Time whenever the program is first detected, so internal code knows it is time to recreate this model
      updateTime: Date.now()
    }
  } );
```
Each model component has a named key, a `type` that refers to the component to be created from the phet-lib namespace, and `args` that hold the arguments for the component constructor.

Within paper programs code, we parse this and a single model from the data. The model itself is observable so that whenever it changes, view and controller components can update/reconstruct/delete.

In this example, the range is meant to be passed to the NumberProperty. This is not set up yet, but is something to figure out.

In team discussions we talked about having many papers to build up a model. But that seems very complicated. Having a single model paper is a good way to start and also may be the best way to construct a model with this tool. We will continue to prototype.

In other programs, you might be able to observe or control the model by referencing these keys.

**Comment by jbphet** [comment]
@jessegreenberg and I experimented yesterday with writing a function in a paper program and passing it through the shared data (which using local storage in the browser) through to the sim design board screen.  Our initial tests with this worked, so it seems like a promising avenue to explore.  In short, we did this:

- Wrote a function in a paper program
- Converted it to string using code like this: `const functionString = myFunction.toString()`
- Used `await paper.set( 'data', ...` to send the string into the shared data space
- Extracted the function string from the shared data space in `board/entry.js`, which is currently the main file of the sim design board
- Evaluated the function using the JS `eval()` function

This was a simple "hello world" sort of test, but it worked, so we will see if we can build on it.
